### PR TITLE
Add basic support for coreclr

### PR DESCRIPTION
**Pull request by gboucher90**

### CODE REVIEW

File: Cottle/src/Builtins/BuiltinFunctions.cs [hunk 1]
```diff
@@ -228,7 +229,11 @@ public static class BuiltinFunctions
 				return -1;
 			}
 			else
-				return source.AsString.IndexOf (search.AsString, offset, StringComparison.InvariantCulture);
+#if CORECLR
+				return source.AsString.IndexOf (search.AsString, offset);
+#else
+				return source.AsString.IndexOf(search.AsString, offset, StringComparison.InvariantCulture);
```
Comment (r3c) [inline Cottle/src/Builtins/BuiltinFunctions.cs]: Please try to comply with existing coding style for this commit (space before opening parenthesis). I'll probably switch to a more common one in a separate commit so future merge request get easier, if any.

File: Cottle/src/Builtins/BuiltinFunctions.cs [hunk 2]
```diff
@@ -254,7 +259,11 @@ public static class BuiltinFunctions
 			int			index;
 			object		target;
 
-			culture = values.Count > 2 ? CultureInfo.GetCultureInfo (values[2].AsString) : CultureInfo.CurrentCulture;
+#if CORECLR
+			culture = values.Count > 2 ? new CultureInfo(values[2].AsString) : CultureInfo.CurrentCulture;
```
Comment (r3c) [inline Cottle/src/Builtins/BuiltinFunctions.cs]: See above comment on space before opening parenthesis.

File: Cottle/src/Values/ScalarValue.cs [hunk 3]
```diff
@@ -66,3 +66,10 @@ public override int GetHashCode ()
 		#endregion
 	}
 }
+
+#if CORECLR
+namespace System
+{
+  public delegate TOutput Converter<TInput, TOutput>(TInput input);
```
Comment (r3c) [inline Cottle/src/Values/ScalarValue.cs]: Please move this converter to its own file.

File: Cottle/src/Builtins/BuiltinFunctions.cs (hunk 1 above)
Comment (gboucher90) [inline Cottle/src/Builtins/BuiltinFunctions.cs]: Yes indeed, sorry I didn't pay attention to the spaces. I updated the commit and hopefully I haven't missed one.

File: Cottle/src/Values/ScalarValue.cs (hunk 3 above)
Comment (gboucher90) [inline Cottle/src/Values/ScalarValue.cs]: Yes, ok, done

**Comment by r3c** [comment]
There are a few left, but they'll disappear when changing CS for the entire project so don't bother. Thanks for the update :)

**Pull request merged**